atef/widgets/core.py

```diff
@@ -0,0 +1,46 @@
+"""
+Core classes for atef Qt-based display GUIs.
+"""
+from typing import ClassVar
+
+from qtpy.uic import loadUiType
+
+from ..util import ATEF_SOURCE_PATH
+
+
+class DesignerDisplay:
+    """Helper class for loading designer .ui files and adding logic."""
+    filename: ClassVar[str]
+
+    def __init_subclass__(cls):
+        """Read the file when the class is created"""
+        super().__init_subclass__()
+        cls.ui_form, _ = loadUiType(
+            str(ATEF_SOURCE_PATH / 'ui' / cls.filename)
+        )
+
+    def __init__(self, *args, **kwargs):
+        """Apply the file to this widget when the instance is created"""
+        super().__init__(*args, **kwargs)
+        self.ui_form.setupUi(self, self)
+
+    def retranslateUi(self, *args, **kwargs):
+        """Required function for setupUi to work in __init__"""
+        self.ui_form.retranslateUi(self, *args, **kwargs)
+
+    def show_type_hints(self):
```

Comment:
This is a good addition for dev too, taking advantage of the copious widget type annotations that are already useful for type checking and IDE completion stuff